src/main/java/online/book/store/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+import online.book.store.repository.book.BookRepository;
+import online.book.store.repository.cartitem.CartItemRepository;
+import online.book.store.repository.shoppingcart.ShoppingCartRepository;
+import online.book.store.service.ShoppingCartService;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final BookRepository bookRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    public void createShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Transactional
+    @Override
+    public ShoppingCartDto getShoppingCartByUser() {
+        return shoppingCartMapper.toDto(findShoppingCartByUser());
+    }
+
+    @Transactional
+    @Override
+    public ShoppingCartDto addBookToShoppingCart(CartItemRequestDto requestDto) {
+        Book book = bookRepository.findById(requestDto.getBookId()).orElseThrow(
+                () -> new EntityNotFoundException(
+                        String.format("Can't find cartItem by id: %s", requestDto.getBookId()))
+        );
+        ShoppingCart shoppingCart = findShoppingCartByUser();
+        checkBookInShoppingCart(shoppingCart.getCartItems(), book.getId());
+        CartItem cartItem = cartItemMapper.toModel(requestDto);
+        cartItem.setShoppingCart(shoppingCart);
+        cartItem.setBook(book);
+        shoppingCart.getCartItems().add(cartItem);
+        shoppingCartRepository.save(shoppingCart);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Transactional
+    @Override
+    public ShoppingCartDto updateCartItem(UpdateCartItemRequestDto requestDto, Long cartItemId) {
+        CartItem cartItem = cartItemRepository.findCartItemById(cartItemId).orElseThrow(
+                () -> new EntityNotFoundException(
+                        String.format("Can't find cartItem by id: %s", cartItemId))
+        );
+        cartItem.setQuantity(requestDto.getQuantity());
+        cartItemRepository.save(cartItem);
+        return shoppingCartMapper.toDto(findShoppingCartByUser());
+    }
+
+    @Override
+    public ShoppingCartDto deleteCartItemById(Long cartItemId) {
+        CartItem cartItem = cartItemRepository.findCartItemByIdAndShoppingCartId(
+                cartItemId, findShoppingCartByUser().getId());
+        cartItemRepository.delete(cartItem);
+        return shoppingCartMapper.toDto(findShoppingCartByUser());
+    }
+
+    private ShoppingCart findShoppingCartByUser() {
```

Comment:
use `findShoppingCartByUserId` and `orElseThrow()`
```suggestion
    private ShoppingCart findShoppingCartByUser(Long userId) {
```